org.adempiere.base/src/org/compiere/model/MAttachment.java

```diff
@@ -79,13 +81,42 @@ public static MAttachment get (Properties ctx, int AD_Table_ID, int Record_ID)
 	 *	@param Record_ID record
 	 *  @param trxName
 	 *	@return attachment or null
+	 * @deprecated Use {@link #get(Properties, int, int, String, String)}
 	 */
 	public static MAttachment get (Properties ctx, int AD_Table_ID, int Record_ID, String trxName)
 	{
-		final String whereClause = I_AD_Attachment.COLUMNNAME_AD_Table_ID+"=? AND "+I_AD_Attachment.COLUMNNAME_Record_ID+"=?";
-		MAttachment retValue = new Query(ctx,I_AD_Attachment.Table_Name,whereClause, trxName)
-		.setParameters(AD_Table_ID, Record_ID)
-		.first();
+		return get(ctx, AD_Table_ID, Record_ID, (String)null, trxName);
+	}	//	get
+	
+	/**
+	 * 	Get Attachment (if there are more than one attachment it gets the first in no specific order)
+	 *	@param ctx context
+	 *	@param AD_Table_ID table
+	 *	@param Record_ID record
+	 *	@param Record_UU record UUID
+	 *  @param trxName
+	 *	@return attachment or null
+	 */
+	public static MAttachment get (Properties ctx, int AD_Table_ID, int Record_ID, String Record_UU, String trxName)
+	{
+		StringBuilder whereClause = new StringBuilder("AD_Table_ID=?");
+		List<Object> params = new ArrayList<Object>();
+		params.add(AD_Table_ID);
+		if (Record_ID > 0) {
+			whereClause.append(" AND Record_ID=?");
+			params.add(Record_ID);
+		}
+		if (!Util.isEmpty(Record_UU)) {
+			whereClause.append(" AND Record_UU=?");
+			params.add(Record_UU);
+		}

```

Comment:
While it is harmless, it seems rather strange that the where clause would include both Record_ID=? and Record_UU=?.